### PR TITLE
Fuse is waiting

### PR DESCRIPTION
**Pull request by edelangh**

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -51,13 +60,47 @@ impl Channel {
...
+            gid: 0,
+            pid: 0,
+            padding: 0
+        };
+        let byte_response: [u8; 40] = unsafe { mem::transmute(response) };
+
+        match errno().0 {
+            libc::EAGAIN => {
+                // this is u32, think about endian
+                unsafe { buffer.set_len(0) };
```
Comment (roblabla) [inline src/channel.rs]: Use buffer.clear() instead, which is safe !

**Comment by roblabla** [comment]
Could you explain what this feature does ? I'm not too sure I understand what's going on there.

**Review by roblabla — CHANGES_REQUESTED** [review]
Could you explain what's going on there ? I'm having trouble piecing everything together.

**Comment by edelangh** [comment]
Thank you roblabla for this point, I just fix it
I find that strange that I cannot directly assign my new Vec to buffer

**Review by roblabla — APPROVED** [review]

**Comment by roblabla** [comment]
Doesn't `*buffer = Vec::new()` work ? Anyway, the clear() is better because it keeps the vec allocated instead of reallocating a new one.

**Comment by roblabla** [comment]
So just for future reference : this PR is deeply tied with our proprietary daemon and basically allows us (in a somewhat dirty way) to get better performances. It's basically impossible to upstream as-is though.

While it certainly is more work, a better way to improve our performances would be fixing https://github.com/zargony/rust-fuse/issues/74 and using `epoll` to avoid threading (and all the mutex and jazz that comes with it).

**Comment by edelangh** [comment]
`*buffer = Vec::new()` doesn't work

**Pull request closed without merging**